inginious/frontend/courses.py

```diff
@@ -80,6 +81,10 @@ def __init__(self, courseid, content, course_fs, task_factory, plugin_manager, t
             self._lti_keys = self._content.get('lti_keys', {})
             self._lti_send_back_grade = self._content.get('lti_send_back_grade', False)
             self._tags = {key: Tag(key, tag_dict, self.gettext) for key, tag_dict in self._content.get("tags", {}).items()}
+            self._additional_fields = {key: AdditionalField(key,
+                                                            field_dict["description"],
+                                                            field_dict["type"])
```

Comment:
You should check here (or in the `AdditionalField` constructor, as this is probably done for `Tag`) that the `field_dict["type"]` value is actually valid and raise an exception if this is not the case.

Currently those are stored in litteral form in the `course.yaml` file. Maybe it's more relevant to store the integer values instead to keep the code more modulable (it is easier to play with the enum numbers in a case of refactor than with the enum name itself)